magento/magento2
# Divide eav attribute cache per website

## Actual pull request -> https://github.com/magento/magento2/pull/34856

### Description (*)
We have an ability to choose different attribute settings per website in Magento admin, but have no per website eav attribute cache separation. So our attribute settings will be depended on which website will be opened first after eav cache cleaning.

### Fixed Issues (if relevant)

1. Fixes magento/magento2#10978

### Manual testing scenarios (*)
1. Open clean Magento instance

Creating new website and store
1. Go to Stores -> Settings -> All Stores
1. Select Create Website
1. Create a new website
1. Select Create Store
1. Create a new store and assign it to a newly created website
1. Select Create Store View
1. Create a new enabled store view and assign it to newly created store
1. Remember store view code
1. Go to System -> Cache Management
1. Flush Cache Storage

Filling required settings
1. Go to Stores -> Configuration -> Customers -> Customer Configuration -> Name and Address Options
1. Select Main Website scope
1. Set Date of Birth to No
1. Set Show Tax/VAT Number to Optional
1. Save config
1. Select new website scope
1. Set Date of Birth to Optional
1. Set Show Tax/VAT Number to No
1. Save config

Enabling store code in urls for fast website switching
1. Select Default Config scope
1. Go to General -> Web -> Url Options
1. Change Add Store Code to Urls to Yes
1. Save config

Reproducing the bug
1. Go to System -> Cache Management
1. Flush cache storage
1. Important! First of all, go to Main Website Storefront Create Account Page
1. Go to https://yourmagento.com/default/customer/account/create/
1. We see Tax/Vat Number field and don't see Dob field
1. Then go to new created website Create Account Page
1. Go to https://yourmagento.com/new-store-view-code/customer/account/create/

Expected:
1. We don't see Tax/Vat Number field and see Dob field

Actual:
1. We see Tax/Vat Number field and don't see Dob field

Then another case:
1. Flush cache storage
1. Important! First of all, go to New Website Storefront Create Account Page
1. Go to yourmagento.com/new-store-view-code/customer/account/create/
1. We don't see Tax/Vat Number field and see Dob field (as expected sooner above)
1. Now go to Main Website Storefront create account page
1. Go to https://yourmagento.com/default/customer/account/create/

Expected:
1. We see Tax/Vat Number field and don't see Dob field

Actual:
1. We don't see Tax/Vat Number field and see Dob field


### Contribution checklist (*)
